tests/PHPStan/Type/CallableTypeTest.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PHPStan\TrinaryLogic;
+
+class CallableTypeTest extends \PHPStan\TestCase
+{
+
+	public function dataIsSubsetOf(): array
+	{
+		return [
+			[
+				new CallableType(),
+				new CallableType(),
+				TrinaryLogic::createYes(),
+			],
+			[
+				new CallableType(),
+				new StringType(),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new IntegerType(),
+				TrinaryLogic::createNo(),
+			],
+			[
+				new CallableType(),
+				new CommonUnionType([new CallableType(), new NullType()]),
+				TrinaryLogic::createYes(),
+			],
+			[
+				new CallableType(),
+				new CommonUnionType([new StringType(), new NullType()]),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new CommonUnionType([new IntegerType(), new NullType()]),
+				TrinaryLogic::createNo(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new CallableType()]),
+				TrinaryLogic::createYes(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new StringType()]),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new IntegerType()]),
+				TrinaryLogic::createNo(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new CallableType(), new StringType()]),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new CallableType(), new ObjectType('Unknown')]),
+				TrinaryLogic::createMaybe(),
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider dataIsSubsetOf
+	 * @param CallableType $callableType
+	 * @param Type $otherType
+	 * @param int $expectedResult
+	 */
+	public function testIsSubsetOf(CallableType $callableType, Type $otherType, TrinaryLogic $expectedResult)
+	{
+		$this->createBroker();
```

Comment:
Is this some hidden side effect? Don't like this :(

Reply:
Well, this is because we don't have TypeFactory (which allowed for using DI) and instead the types use `Broker::getInstance()`. Therefore a broker instance must be created before any Type instance can be used.

Reply:
Yep, I understand and I get that `getInstance()` cannot create Broker instance on its own. OK, fine. 👍